inf-clojure.el

```diff
@@ -1254,15 +1254,43 @@ See variable `inf-clojure-buffer'."
   "Return DATA if and only if it is a list."
   (when (listp data) data))
 
-(defun inf-clojure-completions (expr)
-  "Return a list of completions for the Clojure expression starting with EXPR."
+(defun inf-clojure-list-completions (expr)
+  "Return completions for the Clojure expression starting with EXPR.
+
+Its only ability is to parse a list of candidate strings, every
+other EXPR will be discarded and nil will be returned."
   (when (not (string-blank-p expr))
     (thread-first
         (format (inf-clojure-completion-form) (substring-no-properties expr))
       (inf-clojure--process-response (inf-clojure-proc) "(" ")")
       (inf-clojure--read-or-nil)
       (inf-clojure--list-or-nil))))
 
+(defcustom inf-clojure-completions 'inf-clojure-list-completions
+  "The function that parses completion results.
+
+It is a single-arity function that will receive the REPL
+evaluation result of \\[inf-clojure-completion-form] as string and
+should return elisp data compatible with your completion mode.
+
+The easiest possible data passed in input is a list of
+candidates (e.g.: (\"def\" \"defn\")) but more complex libraries
+like `alexander-yakushev/compliment' can return other things like
+edn.
+
+The expected return depends on the mode that you use for
+completion: `company-mode' for instance allows metadata along
+with the candidates completion. See \\[completion-table-dynamic]
+for more details.
+
+The default value is the `inf-clojure-list-completions' function,
```

Comment:
As the the default function does more than parsing the results (it's also invokes the Clojure completion code) this should probably the adjusted.